library_generation/README.md

```diff
@@ -0,0 +1,113 @@
+
+### proto_path
+A directory in the current working directory (refers as `$cwd`) and copy proto files into it. 
+The absolute path of `proto_path` is `$cwd/$proto_path`. 
+
+Use `-p` or `--proto_path` to specify the value.
+
+### destination_path 
+A directory within `$cwd`. 
+This is the path in which the generated library will reside. 
+The absolute path of `destination_path` is `$cwd/$destination_path`. 
+
+Use `-d` or `--destination_path` to specify the value.
+   
+Note that you do not need to create `$detination_path` beforehand.
+
+The directory structure of the generated library is
+```
+$destination_path
+  |_gapic-*
+  |    |_src
+  |       |_main
+  |          |_java
+  |          |_resources
+  |       |_test
+  |_grpc-*
+  |    |_src
+  |       |_main
+  |          |_java
+  |    
+  |_proto-*
+  |    |_src
+  |       |_main
+  |          |_java
+  |          |_proto
+  |_samples
+      |_snippets
+          |_generated
+```
+You can't build the library as-is since it does not have `pom.xml` or `build.gradle`.
+To use the library, copy the generated files to the corresponding directory
+of a library repository, e.g., `google-cloud-java`.
+
+### gapic_generator_version
+You can find the released version of gapic-generator-java in [maven central](https://repo1.maven.org/maven2/com/google/api/gapic-generator-java/).
+
+Use `--gapic_generator_version` to specify the value.
+
+Note that you can specify a `SNAPSHOT` version as long as you have build the SNAPSHOT of gapic-generator-java in your maven
+local repository.
+
+### protobuf_version (optional)
+You can find the released version of protobuf in [GitHub](https://github.com/protocolbuffers/protobuf/releases/).
+The default value is defined in `gapic-generator-java-pom-parent/pom.xml`.
+
+Use `--protobuf_version` to specify the value.
+
+Note that if specified, the version should be compatible with gapic-generator-java.
+
+### grpc_version (optional)
+You can find the released version of grpc in [maven central](https://repo1.maven.org/maven2/io/grpc/protoc-gen-grpc-java/).
+The default value is defined in `gapic-generator-java-pom-parent/pom.xml`.
+
+Use `--grpc_version` to specify the value.
+
+Note that if specified, the version should be compatible with gapic-generator-java.
+
+### transport (optional)
+One of GAPIC options passed to the generator. The value is either `grpc` or `grpc+rest`.
+The default value is `grpc`.
+
+Use `--transport` to specify the value.
+
+### rest_numeric_enums (optional)
+One of GAPIC options passed to the generator. The value is either `true` or `false`.
+The default value is `true`.
+
+Use `--rest_numeric_enums` to specify the value.
+
+### include_samples (optional)
+Whether generates code samples. The value is either `true` or `false`. 
+The default value is `true`.
+
+Use `--include_samples` to specify the value.
+
+## An example to generate a client library
+```
+library_generation/generate_library.sh \
+-p google/cloud/confidentialcomputing/v1 \
```

Comment:
Can you please add comments or more examples regarding `proto_path` that all protos referenced in the path has to be in the current working directory(assuming my understanding is correct from another comment)?

Reply:
I added an example of generating showcase client.